src/__tests__/vm.test.ts

```diff
@@ -0,0 +1,436 @@
+        },
+        indexJs
+    )
+    const event: PluginEvent = {
+        ...defaultEvent,
+        event: 'attachments',
+    }
+
+    await vm.methods.processEvent(event)
+
+    expect(event.properties).toEqual(attachments)
+})
+
+test('prepareForRun without token', async () => {
+    const indexJs = `
+        async function processEvent (event, meta) {
+            event.properties = {
+                posthog: posthog
+            }
+            return event
+        }
+    `
+    const pluginConfig = { ...mockConfig }
+    const vm = createPluginConfigVM(mockServer, pluginConfig, indexJs)
+    pluginConfig.vm = vm
+    const event: PluginEvent = {
+        ...defaultEvent,
+        event: 'prepareForRun event',
+        properties: {},
+    }
+    const processEvent = prepareForRun(mockServer, pluginConfig.team_id, pluginConfig, 'processEvent', event)
+
+    expect(processEvent).toBeDefined()
+
+    await processEvent!(event)
+
+    expect(event.properties!.posthog).toEqual(null)
+})
+
+test('prepareForRun with token gets posthog', async () => {
+    const indexJs = `
+        async function processEvent (event, meta) {
+            event.properties = {
+                posthog: posthog
+            }
+            return event
+        }
+    `
+    const pluginConfig = { ...mockConfig }
+    const vm = createPluginConfigVM(mockServer, pluginConfig, indexJs)
+    pluginConfig.vm = vm
+    const event: PluginEvent = {
+        ...defaultEvent,
+        event: 'prepareForRun event',
+        properties: {
+            token: 'posthog-token',
+        },
+    }
+    const processEvent = prepareForRun(mockServer, pluginConfig.team_id, pluginConfig, 'processEvent', event)
+    expect(processEvent).toBeDefined()
+
+    await processEvent!(event)
+
+    expect(event.properties!.posthog.capture).toBeDefined()
+    expect(event.properties!.posthog.identify).toBeDefined()
+})
+
+test('posthog.capture', async () => {
+    const indexJs = `
+        async function processEvent (event, meta) {
+            posthog.capture('random-event', { prop: 'value' })
+            return event
+        }
+    `
+    const pluginConfig = { ...mockConfig }
+    const vm = createPluginConfigVM(mockServer, pluginConfig, indexJs)
+    pluginConfig.vm = vm
+    const event: PluginEvent = {
+        ...defaultEvent,
+        event: 'prepareForRun event',
+        properties: {
+            // needs a token in the event
+            token: 'posthog-token',
+        },
+    }
+    const processEvent = prepareForRun(mockServer, pluginConfig.team_id, pluginConfig, 'processEvent', event)
+    expect(processEvent).toBeDefined()
+    await processEvent!(event)
+
+    expect((fetch as any).mock.calls[0][0]).toContain('http://localhost/e/?ip=1&_=')
+    expect((fetch as any).mock.calls[0][1].body).toContain('data=')
+    expect((fetch as any).mock.calls[0][1].body).toContain('&compression=lz64')
+    expect((fetch as any).mock.calls[0][1]).toMatchObject({
```

Comment:
I think this would be better as `fetch as unknown`.

Reply:
TypeScript is even less happy with that:
![image](https://user-images.githubusercontent.com/53387/100762310-1c9cb180-33f4-11eb-93f8-886443b4e21e.png)

And that's a TS error not ESLint suggestion.

I added a few lines to `eslintrc` to ignore `any`s under tests.

Also, these variables are technically `known`, I'm just ignoring them for convenience... and thus `any` seems like a better fit.